identity-apps-core/components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/util/AppPortalUtils.java

```diff
@@ -127,10 +128,22 @@ public static void createOAuth2Application(String applicationName, String portal
             portalPath = "/" + portalPath;
         }
         String callbackUrl = IdentityUtil.getServerURL(portalPath, true, true);
+        String consoleCallbackUrl = IdentityUtil.getServerURL(portalPath, true, true);
+        String appendedConsoleCallBackURLRegex = StringUtils.EMPTY;
         try {
             // Update the callback URL properly if origin is configured for the portal app.
             callbackUrl = ApplicationMgtUtil.replaceUrlOriginWithPlaceholders(callbackUrl);
             callbackUrl = ApplicationMgtUtil.resolveOriginUrlFromPlaceholders(callbackUrl, applicationName);
+
+            // Add console url when impersonation is enabled.
+            boolean isUserSessionImpersonationEnabled = IdentityUtil.getProperty(USER_SESSION_IMPERSONATION) != null &&
+                Boolean.parseBoolean(IdentityUtil.getProperty(USER_SESSION_IMPERSONATION));
+            if (isUserSessionImpersonationEnabled && MYACCOUNT_APP.equals(applicationName)) {
+                consoleCallbackUrl = ApplicationMgtUtil.replaceUrlOriginWithPlaceholders(consoleCallbackUrl);
+                consoleCallbackUrl = ApplicationMgtUtil.resolveOriginUrlFromPlaceholders(consoleCallbackUrl, CONSOLE_APP);
```

Comment:
can't we do this in one line?

```suggestion
                consoleCallbackUrl = ApplicationMgtUtil.resolveOriginUrlFromPlaceholders(consoleCallbackUrl, CONSOLE_APP);
```